pkg/metrics/metrics.jsx

```diff
@@ -867,6 +872,91 @@ class MetricsHour extends React.Component {
+
+const PCPConfig = ({ buttonVariant }) => {
+    const [dialogVisible, setDialogVisible] = useState(false);
+    const [dialogError, setDialogError] = useState(null);
+    const [dialogLoggerValue, setDialogLoggerValue] = useState(false);
+    const [pending, setPending] = useState(false);
+    const s_pmlogger = useObject(() => service.proxy("pmlogger.service"), null, []);
+
+    debug("PCPConfig s_pmlogger.state", s_pmlogger.state);
+
+    useEvent(s_pmlogger, "changed");
+    useEvent(superuser, "changed");
+
+    if (!superuser.allowed)
+        return null;
+
+    const handleApply = () => {
+        setPending(true);
+        if (dialogLoggerValue) {
+            Promise.all([s_pmlogger.start(), s_pmlogger.enable()])
+                    .then(() => setDialogVisible(false))
+                    .catch(err => setDialogError(err.toString()))
+                    .finally(() => setPending(false));
+        } else {
+            Promise.all([s_pmlogger.stop(), s_pmlogger.disable()])
+                    .then(() => setDialogVisible(false))
+                    .catch(err => setDialogError(err.toString()))
+                    .finally(() => setPending(false));
+        }
+    };
+
+    return (
+        <>
+            <Button variant={buttonVariant} icon={<CogIcon />}
+                    isDisabled={ !persistentServiceState(s_pmlogger) }
+                    onClick={ () => {
+                        setDialogLoggerValue(s_pmlogger.state === 'running');
+                        setDialogVisible(true);
+                    } }>
+                { _("Metrics settings") }
+            </Button>
+
+            {dialogVisible &&
+            <Modal position="top" variant="small" isOpen
+                   id="pcp-settings-modal"
+                   onClose={ () => setDialogVisible(false) }
+                   title={ _("Metrics settings") }
+                   footer={<>
+                       { dialogError && <ModalError dialogError={ _("Failed to configure PCP") } dialogErrorDetail={dialogError} /> }
+
+                       <Button variant='primary' onClick={handleApply} isDisabled={pending}>
+                           { _("Apply") }
+                       </Button>
+                       <Button variant='link' className='btn-cancel' onClick={ () => setDialogVisible(false) }>
+                           {_("Cancel")}
+                       </Button>
+                   </>
+                   }>
+
+                <Text component={TextVariants.p} className="pcp-settings-modal-text">
+                    { _("Performance Co-Pilot collects and analyzes performance metrics from your system.") }
+
+                    <a href="https://pcp.io/" target="_blank" rel="noopener noreferrer">
+                        <ExternalLinkAltIcon />&nbsp;{ _("Read more...") }
+                    </a>
+                </Text>
+
+                <Flex direction={{ default: 'column' }}>
```

Comment:
Using the flex here results in misalignments.
![Screen Shot 2021-05-19 at 12 44 51](https://user-images.githubusercontent.com/14921356/118638216-e18e0f00-b7d6-11eb-8115-8531918fc89a.png)
 

I will try to find a better apporach.

Reply:
In addition the Switch should have a label or an aria label to be accessible. Clicking on the label should also enable/disable the switch.

Reply:
I could also use the "builtin" label of a switch, which will most probably fix the alignment, and certainly fix the "linking" of the two. The main downside is that the label becomes light grey when the button is off, and thus the widget looks disabled when it's not. But that's a general problem of the switches, so let's rather fix that with CSS if we care. Then I just need to have a way to add the (unitname) bit.

I'll make a version with that, and re-post.

Reply:
Right, that should be fixed globally is patternfly overrides file if we decide it's wrong.

Reply:
Thanks for your fixup! applied.